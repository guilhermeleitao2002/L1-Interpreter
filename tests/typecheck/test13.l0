type Circle = struct { #cx:int, #cy:int, #rad:int };
type Rectangle = struct { #h:int, #w:int };
type Blob = union { #circle: Circle, #rect: Rectangle };
let pi = 3142;
let area = fn b:union{ #rect: Rectangle } => {
  match b {
        #circle(c) -> let r = pi*c.#rad; (r*r)/2
    |   #rect(r) -> r.#h * r.#w
    |   #something(_) -> 0
    }
};
let r1:Blob = #rect({ #h = 10, #w = 20 });
area(r1);;