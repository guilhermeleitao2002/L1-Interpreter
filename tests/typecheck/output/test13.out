TypeError: Case circle not found in union type